Make reproducible builds from master branch

## Description

Every time I try to [build](https://github.com/software-mansion/react-native-reanimated/issues/950) **react-native-reanimated** npm package from master it fails. 

I use script `createNPMPackage.sh`, other ways didn't work. Being able to build complete npm package from master is very important, otherwise I can't test patches against master branch and can't verify they are working.

```bash
...
> Task :buildReanimate
...
make: Leaving directory '/home/me/react-native-reanimated/android/src/main/JNI'
/home/me/react-native-reanimated/android/build/third-party-ndk/folly/folly/FileUtil.cpp:37:14: error: no matching function for call to 'wrapNoInt'
  return int(wrapNoInt(open, name, flags, mode));
             ^~~~~~~~~
/home/me/react-native-reanimated/android/build/third-party-ndk/folly/folly/detail/FileUtilDetail.h:34:9: note: candidate template ignored: couldn't infer template argument 'F'
ssize_t wrapNoInt(F f, Args... args) {
        ^
1 error generated.
make: *** [/home/me/.local/opt/android-sdk/ndk-bundle/build/core/build-binary.mk:478: /home/me/react-native-reanimated/android/build/tmp/buildReanimated/local/arm64-v8a/objs/folly_json/folly/FileUtil.o] Error 1
> Task :buildReanimated FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':buildReanimated'.
> Process 'command '/home/me/.local/opt/android-sdk/ndk-bundle/ndk-build'' finished with non-zero exit value 2
```

It would be very good to have CI tests that check source code from master at least is building for iOS and Android.

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -86,6 +86,14 @@ def dependenciesPath = System.getenv("REACT_NATIVE_DEPENDENCIES")
 // and the build will use that.
 def boostPath = dependenciesPath ?: System.getenv("REACT_NATIVE_BOOST_PATH")
 
+def follyReplaceContent = '''
+  ssize_t r;
+  do {
+    r = open(name, flags, mode);
+  } while (r == -1 && errno == EINTR);
+  return r;
+'''
+
 task Log {
     print("building Reanimated2")
 }
@@ -141,6 +149,9 @@ task prepareFolly(dependsOn: dependenciesPath ? [] : [downloadFolly], type: Copy
     from("$reactNativeThirdParty/folly/Android.mk")
     include("folly-${FOLLY_VERSION}/folly/**/*", "Android.mk")
     eachFile { fname -> fname.path = (fname.path - "folly-${FOLLY_VERSION}/") }
+    // Fixes problem with Folly failing to build on certain systems. See
+    // https://github.com/software-mansion/react-native-reanimated/issues/1024
+    filter { line -> line.replaceAll('return int\\(wrapNoInt\\(open, name, flags, mode\\)\\);', follyReplaceContent) }
     includeEmptyDirs = false
     into("$thirdPartyNdkDir/folly")
 }

